### DISCUSSION TITLE
API to get native device stream pointer

### DISCUSSION THREAD

**Discussion by deukhyun-cha** [libocca/occa#659] [General]
We have a case in our development that a 3rd party library requires device stream pointer to be passed in to one of its APIs - stream might be one of the most common elements to be communicated between the modules as it requires a global management. Now OCCA has nicely implemented stream related APIs including _wrapStream_ (although it seems only available through cpp APIs at the moment) which could provide an alternative for the situation. Only drawback of that is to introduce device native calls (CUDA or HIP for instance) in the application where all the other offloading implementations are nicely encapsulated by OCCA. If we have an OCCA API that returns native device stream pointer as a void* for the current stream set for any platform it's running on, it could help to minimize the platform-specific dependency in the application and keep it in OCCA layer.

**Comment by stgeke** [comment]
https://github.com/libocca/occa/pull/635

**Reply by deukhyun-cha** [reply]
Thanks for the comment. I might be mistaken, but it doesn't seem to have a C API for this, does it?

**Reply by deukhyun-cha** [reply]
@stgeke and @kris-rowe, sorry for poking on this, should I create a new issue for the C API for this feature if it doesn't exist?

**Reply by kris-rowe** [reply]
Correct. I don't think it has propagated into the C or Fortran API. Please do open an issue.

**Reply by deukhyun-cha** [reply]
Thank you @kris-rowe. Just opened an issue #664.
